be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -166,6 +167,24 @@ void ColumnReader::get_row_ranges_by_zone_map(CondColumn* cond_column, RowRanges
     _calculate_row_ranges(page_indexes, row_ranges);
 }
 
+Status ColumnReader::get_dict_page_decoder(BinaryDictPageDecoder* binaryDictPageDecoder) {
```

Comment:
should return _column_dict_page_decoder, not set _column_dict_page_decoder in this function

Reply:
return status here can make FileColumnIterator returns error when read_page failed or dict_page_decoder-> init failed;

Reply:
ColumnReader is shared by multiple FileColumnIterator and thus should be thread-safe.

Reply:
👌